pkg/app/piped/driftdetector/ecs/detector.go

```diff
@@ -316,6 +327,12 @@ func ignoreParameters(liveManifests provider.ECSManifests, headManifests provide
 			// Essential is true by default. See https://docs.aws.amazon.com/AmazonECS/latest/APIReference/API_ContainerDefinition.html#ECS-Type-ContainerDefinition-es.
 			cd.Essential = aws.Bool(true)
 		}
+
+		// Sort the head environment variables by name.
+		envs := slices.Clone(cd.Environment)
+		envs = sortKeyPairs(envs)
+		cd.Environment = envs
```

Comment:
After refactoring `sortKeyPairs()`

```suggestion
		cd.Environment = sortKeyPairs(cd.Environment)
```